### ISSUE TITLE
I have a query about the rollup build process with export classes as opposed to export functions

### ISSUE THREAD

**Issue by malee1975** [rollup/rollup#5516]
I have noticed that when rollup is used to compile and that if a function is defined within an export class. The entire class and all of its functions is included within the distributable JS. Is there some kind of tree-shaking that can be applied to allow only functions used within a class to be incorporated into the built JS?

**Comment by liuly0322** [comment]
#349 

> @wmhilton Currently, the only JavaScript optimizer (that I am aware of) that can do dead code elimination on class methods is the [Google Closure Compiler](https://github.com/google/closure-compiler). It does a lot of awesome optimizations that other tools (e.g. Webpack, Uglify, Rollup) do not do.
> However, in order to achieve those optimizations, it requires you to [follow various restrictions](https://developers.google.com/closure/compiler/docs/limitations). Those restrictions means that most JavaScript libraries won't work. Generally, only code which has been specifically written for Google Closure Compiler will work.

As JavaScript is a highly dynamic language, it is hard to decide if meta info of a class will be used, or if the class will be passed as a whole and used by other unknown methods, unlike "top-level" function themselves can be easily identified all uses.

> Maybe farer future. This would be rather complex to get right, object expression tree-shaking should probably come first, and that one is currently far down the pipeline

#5420 may achieve object expression tree-shaking! So maybe in future, the class method tree-shaking will be supported.

**Comment by lukastaegert** [comment]
Exactly! I still need to find some time to properly review it, but the goal is to at least lay the groundwork for this. Once we can control object properties, we can start looking into the additional complexities of classes.

**Comment by malee1975** [comment]
Sounds good :)
